mlir/lib/Dialect/ArmSME/Transforms/TileAllocation.cpp

```diff
@@ -137,6 +137,8 @@ static ArrayRef<TileMask> getMasks(ArmSMETileType type) {
   case ArmSMETileType::ZAQ:
     return ZA_Q_MASKS;
   }
+  assert("unknown type in getMasks" == nullptr);
+  return {};
```

Comment:
Can't this (and elsewhere) use `llvm_unreachable`? 
```suggestion
  llvm_unreachable("unknown type in getMasks"");
```

Reply:
sure, will do.

Reply:
done